docs/integrations/mirroring.md

```diff
@@ -0,0 +1,89 @@
+
+## Useful objects to use in your code
+All the functions explained here are globally available through the CommonServerPython file.
+
+### get-remote-data
+* GetRemoteDateArgs - this is an object created to maintain all the argument you receive from the server in order to use this command.
+Arguments explanation:
+  - remote_incident_id - the remote incident id.
+  - last_update - the time the incident was last updated - our recommendation would be to use the arg_to_timestamp function to parse it into a timestamp.
+* GetRemoteDataResponse - this is the object that maintains the format in which you should order the results from this function. You should use return_results on this object to make it work.
+Arguments explanation:
+  - mirrored_object - this is essentially the object(dict) of whatever you are trying to mirror - incident in most cases.
+  - entries - a list of entries to add to your incident - for instance:
+```buildoutcfg
+{
+    'Type': entry.get('type'),
+    'Category': entry.get('category'),
+    'Contents': entry.get('contents'),
+    'ContentsFormat': entry.get('format'),
+    'Tags': entry.get('tags'),  # the list of tags to add to the entry
+    'Note': entry.get('note')  # boolean, True for Note, False otherwise
+}
+```
+
+### update-remote-system
+* UpdateRemoteSystemArgs - this is an object created to maintain all the arguments you receive from the server in order to use this command.
+Arguments explanation:
+  - data - will represent the data of the current incident.
+  - entries - will represent the entries from your current incident 
+  - remote_incident_id - the remote incident id.
+  - inc_status - the status of the incident(numeric value, could be used with IncidentStatus from CommonServerPython)
+  - delta - will represent the dictionary of fields which have changed from the last update.
+
+### update-remote-system
+* SchemeTypeMapping - this is the object you should use to keep the correct structure of the mapping for the fetched incident type.
+* GetMappingFieldsResponse - this is the object to gather all your SchemeTypeMapping and then to parse them properly into the results using the return_results command.
+
+## ￿Get going on XSOA￿R incident
+There are few incident fields you need to configure to get going on the system.
+* dbotMirrorDirection - Both or In
+* dbotMirrorId - represents the id of the incident in the external system
+* dbotMirrorInstance - the instance that will perform the mirroring
+
+Useful fields:
+* dbotMirrorTags - tags for mirrored-out entries (comment/files).
+
+## Debugging
+
+* getMirrorStatistics command - a hidden command that returns mirroring statistics: total mirrored, rate limited, and last run.
+* getSyncMirrorRecords - a hidden command that returns records that hold internal mirroring metadata for each mirrored incident
```

Comment:
explain how to debug/troubleshoot get-remote-data and update-remote-system commands.